File: Matlab/MA/ma_stock.m
clc;
clear all;
close all;

data = csvread("..\..\Data\SP500.csv", 2, 1);

adj = data(:,5);

figure
plot(adj)
grid on

ret = diff(log(adj));

figure
plot(ret)
grid on

e = diff(ret);

%Ident
phi = [e(2:end-1) e(1:end-2)];
sys_hat = phi\ret(3:end-1);

y_hat = zeros(length(ret), 1);
y_hat(1:2) = 1;
for k=3:length(ret)
    y_hat(k) = sys_hat'*[e(k-1); e(k-2)];
end

plot(y_hat)

figure
autocorr(ret-y_hat)

File: R/MA/ma_stock.R
#this is how we can fetch finance related date from the web
require(quantmod)

PETR <- read.csv(file="..\\Data\\SP500.csv", header=TRUE, sep=",")

getSymbols("AAPL", src= "yahoo")

adClose <- PETR$Adj.Close
#we want to plot the adjusted closing prices
plot(Ad(AAPL))

#log daily returns of AAPL stock (for adjusted closing price)
aaplreturn = diff(log(Ad(AAPL)))

plot(aaplreturn)
acf(aaplreturn,na.action=na.omit)

#we set AR=0 and integrated component=0 so it is a MA(3)
aaplreturn.ma <- arima(aaplreturn,order=c(0,0,3))
aaplreturn.ma

#drop the first value because it is NA
aaplreturn.ma$res[-1]

#y=x-x' the residual series: if the model is good there is no serial correlation in y(t)
acf(aaplreturn.ma$res[-1])
